src/features/embalm/stepContent/hooks/useCreateSarcohpagus.ts

```diff
@@ -85,49 +96,129 @@ export function useCreateSarcophagus() {
-      const encryptedShards = await encryptShards(archPublicKeys, shards);
+      const encShards = await encryptShards(archPublicKeys, shards);
 
       // Step 3: Upload the encrypted shards mapping to the arweave bundlr
-      const mapping: Record<string, string> = encryptedShards.reduce(
+      const mapping: Record<string, string> = encShards.reduce(
         (acc, shard) => ({
           ...acc,
           [shard.publicKey]: shard.encryptedShard,
         }),
         {}
       );
 
-      const encryptedShardsTxId = await uploadArweaveFile(Buffer.from(JSON.stringify(mapping))); // TODO: change to use uploadFile for Bundlr, once local testing figured out
+      const txId = await uploadArweaveFile(Buffer.from(JSON.stringify(mapping))); // TODO: change to use uploadFile for Bundlr, once local testing figured out
 
-      dispatch(setShardPayloadData(encryptedShards, encryptedShardsTxId));
-      return { encryptedShards, encryptedShardsTxId };
+      setArchaeologistShards(encShards);
+      setEncryptedShardsTxId(txId);
     } catch (error) {
       console.error(error);
-      return emptyData;
-    } finally {
-      dispatch(setIsUploading(false));
     }
+  }, [requiredArchaeologists, outerPrivateKey, selectedArchaeologists, uploadArweaveFile]);
+
+  const uploadAndSetDoubleEncryptedFile = useCallback(async () => {
+    const payload = await readFileDataAsBase64(file!);
+
+    // Step 1: Encrypt the inner layer
+    const encryptedInnerLayer = await encrypt(recipientState.publicKey, payload);
+
+    // Step 2: Encrypt the outer layer
+    const encryptedOuterLayer = await encrypt(outerPublicKey!, encryptedInnerLayer);
+
+    // Step 3: Upload the double encrypted payload to the arweave bundlr
+    // TODO: change to use uploadFile for Bundlr, once local testing figured out
+    const payloadTxId = await uploadArweaveFile(encryptedOuterLayer);
+
+    setSarcophagusPayloadTxId(payloadTxId);
   }, [
-    canCreateSarcophagus,
-    requiredArchaeologists,
-    outerPrivateKey,
-    selectedArchaeologists,
+    file,
+    outerPublicKey,
+    recipientState.publicKey,
     uploadArweaveFile,
-    dispatch,
+    setSarcophagusPayloadTxId,
   ]);
 
-  const handleCreate = useCallback(async () => {
-    // Step 8: Create the sarcophagus
+  // TODO -- add approval stage
+  useEffect(() => {
+    (async () => {
+      const incrementStage = (): void => {
+        const currentIndex = createSarcophagusStages.indexOf(currentStage);
+        setCurrentStage(createSarcophagusStages[currentIndex + 1]);
+      };
+
+      const executeStage = async (stageToExecute: Function, ...args: any[]): Promise<any> => {
+        return new Promise((resolve, reject) => {
+          setStageExecuting(true);
+
+          stageToExecute(...args)
+            .then((result: any) => {
+              setStageExecuting(false);
+
+              // Set current stage to next stage
+              incrementStage();
```

Comment:
I'm curious but haven't tested... Does anything bad happen when `incrementStage()` is called on the final stage?

Reply:
@knoll3 -- good question -- the final stage doesn't have an `executeStage` call (since its just a `COMPLETED` stage), so `incrementStage` will never get called there.